File: benchmark/c/cpw/count_set_bits_in_an_integer/count_set_bits_in_an_integer.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



unsigned int count_set_bits_in_an_integer ( unsigned int n ) ;
unsigned int count_set_bits_in_an_integer ( unsigned int n ) {
  unsigned int count = 0;
  while ( n ) {
    count += n & 1;
    n >>= 1;
  }
  return count;
}

File: translation_result/else_bms/success/c/cpw/count_set_bits_in_an_integer/mistral/CAPR/count_set_bits_in_an_integer.rs


use std::u32;

fn count_set_bits_in_an_integer(n: u32) -> u32 {
    let mut n_local = n;
    let mut count = 0;
    while n_local > 0 {
        count += n_local as u32 & 1;
        n_local = n_local.wrapping_shr(1);
    }
    count
}

